### PR TITLE
feat: change limit character for 12c02

### PR DESCRIPTION
**Pull request by ErDiabIo**
Configurable character limit for 12c02

### CODE REVIEW

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
Changes look good but the tests are failing. Can you please fix it?

**Comment by yajra** [comment]
test

**Comment by yajra** [comment]
Tried to run the PR locally and got the same error:

```
BadMethodCallException: Method Yajra\Oci8\Query\Grammars\OracleGrammar::setMaxLength does not exist.
```

Converting the PR to draft until we fix the tests. Thanks!

**Review by yajra — APPROVED** [review]

**Comment by yajra** [comment]
@ErDiabIo just to be sure, this is targeted on Laravel 8 right?

**Comment by ErDiabIo** [comment]
@yajra Yes, is targeted on Laravel 8.

File: src/config/oracle.php [hunk 1]
```diff
@@ -16,6 +16,7 @@
         'edition'        => env('DB_EDITION', 'ora$base'),
         'server_version' => env('DB_SERVER_VERSION', '11g'),
         'load_balance'   => env('DB_LOAD_BALANCE', 'yes'),
+        'max_name_len'   => env('ORA_MAX_NAME_LEN', 30),
```
Comment (yajra) [inline src/config/oracle.php:19]: Maybe we can use the `server_version` to identify the length like on PR https://github.com/yajra/laravel-oci8/pull/768?

`>=12` will set the length to 126 characters.

**Review by yajra — CHANGES_REQUESTED** [review]
I suggest we use the `server_version` to identify the max length.

Other than that, I tested the PR on a new Laravel 8 and it works well.

File: src/Oci8/Oci8Connection.php [hunk 2]
```diff
@@ -367,6 +368,16 @@ protected function getConfigSchemaPrefix()
...
     }
 
+    /**
+     * Get config max length.
+     *
+     * @return string
+     */
+    protected function getConfigMaxLength()
+    {
+        return isset($this->config['max_name_len']) ? $this->config['max_name_len'] : 30;
```
Comment (yajra) [inline src/Oci8/Oci8Connection.php:378]: Maybe something like on PR #768 

```php
$version = (int) filter_var($this->connection->getConfig('server_version'), FILTER_SANITIZE_NUMBER_INT);

return $version >= 12 ? 126 : 30;
```

Comment (ErDiabIo) [reply src/Oci8/Oci8Connection.php:378]: So, somerhing like this ?
```php
$version = (int) filter_var($this->config['server_version'], FILTER_SANITIZE_NUMBER_INT);
return $version >= 12 ? $this->config['max_name_len'] ?? 30 : 30;
```

Comment (yajra) [reply src/Oci8/Oci8Connection.php:378]: No need for `max_name_len`. I think 126 is the new constant length for 12 and above.

Comment (yajra) [reply src/Oci8/Oci8Connection.php:378]: We can also remove the max_name_len config added in the PR.

Comment (ErDiabIo) [reply src/Oci8/Oci8Connection.php:378]: > No need for `max_name_len`. I think 126 is the new constant length for 12 and above.
For me is ok, but they change the limit in oracle 12.2.0.1 if we compare with 12 some people can have a problem if they have 12.1.0.1

Comment (yajra) [reply src/Oci8/Oci8Connection.php:378]: Updated the suggestion to reflect your code on config.

Comment (yajra) [reply src/Oci8/Oci8Connection.php:378]: That makes sense. Also reading the new max length, is 128 per [SO post](https://stackoverflow.com/questions/756558/what-is-the-maximum-length-of-a-table-name-in-oracle)

> In [Oracle 12.2](http://docs.oracle.com/database/122/REFRN/ALL_OBJECTS.htm#REFRN20146) and above the maximum object name length is 128 bytes.
> In [Oracle 12.1](https://docs.oracle.com/database/121/SQLRF/sql_elements008.htm#SQLRF00223) and below the maximum object name length is 30 bytes.

**Comment by yajra** [comment]
Released on v8.7.0 🚀 Thanks!

**Pull request merged**